hibernate-core/src/main/java/org/hibernate/cfg/CopyIdentifierComponentSecondPass.java

```diff
@@ -79,8 +79,8 @@ public void doSecondPass(Map persistentClasses) throws MappingException {
 		AtomicInteger index = new AtomicInteger( 0 );
 		if ( columnByReferencedName.isEmpty() ) {
 			isExplicitReference = false;
-			for (Ejb3JoinColumn joinColumn : joinColumns) {
-				columnByReferencedName.put( "" + index.get(), joinColumn );
+			for ( Ejb3JoinColumn joinColumn : joinColumns ) {
+				columnByReferencedName.put( "0", joinColumn );
```

Comment:
the index is incremented inside the for loop, if `joinColumns`contains more than 1 `Ejb3JoinColumn` with this change we are not using the correct value for the Map key.

Reply:
I think the change should be `columnByReferencedName.put(  String.valueOf( index.get() ) , joinColumn );`

Reply:
nice catch. Not sure how I missed the loop.

Reply:
fixed